base/tuple.jl

```diff
@@ -160,7 +160,8 @@ heads(t::Tuple, ts::Tuple...) = (t[1], heads(ts...)...)
 tails() = ()
 tails(t::Tuple, ts::Tuple...) = (tail(t), tails(ts...)...)
 map(f, ::Tuple{}, ts::Tuple...) = ()
-map(f, t1::Tuple, t2::Tuple, ts::Tuple...) = (f(heads(t1, t2, ts...)...), map(f, tails(t1, t2, ts...)...)...)
+map(f, t1::Tuple, t2::Tuple, ts::Tuple...) = (@_inline_meta;
```

Comment:
You're right, I looked at this backwards, is `heads` and `tails` what should be fine to inline. I'll try another solution.